clients/src/main/java/org/apache/kafka/common/security/oauthbearer/internals/secured/ConfigurationUtils.java

```diff
@@ -209,6 +209,19 @@ public String validateString(String name, boolean isRequired) throws ValidateExc
         return value;
     }
 
+    /**
+     * Validates that a value, if supplied, is a {@link Boolean}. If no value is present in the configuration, a
+     * default value of {@link Boolean#FALSE} is returned.
+     */
+    public Boolean validateBoolean(String name) {
```

Comment:
maybe it would be better to pass default value as an argument to this method and change to something like this:

```java
cu.validateBoolean(SASL_OAUTHBEARER_HEADER_URLENCODE, DEFAULT_SASL_OAUTHBEARER_HEADER_URLENCODE);
```

Reply:
I opted to add a separate method in `AccessTokenRetrieverFactory` since the `ConfigurationUtils` class didn't have providing the default as a general pattern.